File: past-competitions/2018/src/mike-java/Prob11.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

public class Prob11 {
    // change this file name for each problem!
    private static final String INPUT_FILE_NAME = "Prob11.in.txt";
    
    public static void main(String[] args) {
        try {
            // prepare to read the file
            File inFile = new File(INPUT_FILE_NAME);
            FileReader fr = new FileReader(inFile);
            BufferedReader br = new BufferedReader(fr);
            String inLine = null;
            
            int[] values = new int[10];
            
            // get the number of test cases
            int T = Integer.parseInt(br.readLine());
            
            // outer loop through test cases
            while (T-- > 0) {
                // BEGINNING OF TEST CASE CODE
                
                // read the next line of text
                inLine = br.readLine();
                
                // split the line
                String[] tokens = inLine.split(" ");
                
                // get the values
                for (int i=0; i<10; i++) {
                    values[i] = Integer.parseInt(tokens[i]);
                }
                
                // find distance between foreground and chroma key
                int intDist = 0;
                for (int i=0; i<3; i++) {
                    int tempDist = values[i+4] - values[i];
                    tempDist = tempDist * tempDist;
                    intDist += tempDist;
                }
                double dist = Math.sqrt(intDist);
                
                // check for tolerance
                double tolerance = (double) values[3];
                
                int index = 4;
                if (dist < tolerance) {
                    index += 3;
                }
                
                // print values
                for (int i=0; i<3; i++) {
                    if (i > 0) System.out.print(" ");
                    System.out.print(values[i+index]);
                }
                System.out.println();
                
                // END OF TEST CASE CODE
            }
            
            // clean up
            br.close();
            fr.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: past-competitions/2018/src/brett-python/Prob11.py
import math
# Open the input file
with open("Prob11.in.txt", "rt") as inputFile:
    # Read the number of test cases (trim out the newline)
    cases = int(inputFile.readline().replace("\n", ""))
    # For each test case
    for caseNum in range(cases):
        # Read the line
        line = inputFile.readline().replace("\n", "")
        # split up the data
        # 0-2 are chroma key, 3 is tolerance,
        # 4-6 are foreground, 7-9 are background
        data = line.split(" ")
        # calculate distance (** 2 squares values)
        redDist = (int(data[0]) - int(data[4])) ** 2
        greenDist = (int(data[1]) - int(data[5])) ** 2
        blueDist = (int(data[2]) - int(data[6])) ** 2
        distance = math.sqrt(redDist + greenDist + blueDist)
        if distance <= int(data[3]):
            print(data[7] + " " + data[8] + " " + data[9])
        else:
            print(data[4] + " " + data[5] + " " + data[6])
